### PR TITLE
WIP: Shapenet segmentation model

### PR DESCRIPTION
**Pull request by soumik12345**
- [x] Segmentation model for shape part segmentation
- [x] Corresponding testcases
- [x] Run testcases inside docker container

### CODE REVIEW

**Review by sayakpaul — APPROVED** [review]

File: point_seg/pointnet/models.py [hunk 1]
```diff
@@ -45,3 +49,43 @@ def get_baseline_segmentation_model(num_points: int, num_classes: int) -> keras.
     return keras.Model(
         input_points, outputs, name="baseline_pointnet_segmentation_model"
     )
+
+
+def get_shape_segmentation_model(num_points: int, num_classes: int) -> keras.Model:
+    input_points = keras.Input(shape=(num_points, 3))
+
```
[the remark this answers is not in the document]
Comment (soumik12345) [reply point_seg/pointnet/models.py:56]: @sayakpaul Yes.

**Comment by sayakpaul** [comment]
We can merge the PR upon successful competition of the action.

File: point_seg/pointnet/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sayakpaul) [reply point_seg/pointnet/models.py:56]: But that also includes a one-hot encoded vector concatenated right?

**Comment by sayakpaul** [comment]
> The result of concatenation in the figure has a dimension of 3024 which in our implementation is coming as 3008.
What is meant by one-hot in the given figure?

Unfortunately, I am also doubtful about these two questions. Their original implementation also does not yield a 3024-d vector after the concatenation block if I read the code correctly.

**Pull request merged**